### PR TITLE
Enforce Return

### PR DESCRIPTION
**Pull request by geggleto**
- Adds a new option to enforce return's from route actions

### CODE REVIEW

File: Slim/Exception/BadRouteReturnException.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
+/**
```
Comment (dopesong) [inline Slim/Exception/BadRouteReturnException.php]: Copyright?

File: Slim/Route.php [hunk 2]
```diff
@@ -207,6 +214,26 @@ public function setOutputBuffering($mode)
...
+    {
+        return $this->enforceReturn;
+    }
+
+    /**
+     * @param boolean $enforceReturn
+     */
+    public function setEnforceReturn($enforceReturn)
+    {
+        $this->enforceReturn = $enforceReturn;
```
Comment (akrabat) [inline Slim/Route.php]: Cast to a `bool` please.

File: Slim/Route.php [hunk 3]
```diff
@@ -9,6 +9,8 @@
 namespace Slim;
 
 use Exception;
+use FastRoute\BadRouteException;
```
Comment (akrabat) [inline Slim/Route.php]: Not needed.

File: Slim/Route.php [hunk 4]
```diff
@@ -65,6 +67,11 @@ class Route extends Routable implements RouteInterface
     protected $outputBuffering = 'append';
 
     /**
+     * @var bool
```
Comment (akrabat) [inline Slim/Route.php:71]: Needs title.

File: Slim/Route.php [hunk 5]
```diff
@@ -207,6 +214,26 @@ public function setOutputBuffering($mode)
...
+
+    /**
+     * @param boolean $enforceReturn
+     */
+    public function setEnforceReturn($enforceReturn)
+    {
+        $this->enforceReturn = $enforceReturn;
+    }
+
+
```
Comment (akrabat) [inline Slim/Route.php]: Too many blank lines here.

File: Slim/Route.php [hunk 6]
```diff
@@ -65,6 +67,11 @@ class Route extends Routable implements RouteInterface
     protected $outputBuffering = 'append';
 
     /**
+     * @var bool
+     */
+    protected $enforceReturn = false;
```
Comment (akrabat) [inline Slim/Route.php]: Rename to `enforceReturnOfResponse` ?

File: tests/RouteTest.php [hunk 7]
```diff
@@ -453,4 +454,33 @@ public function testChangingCallable()
...
+     * @expectedException \Slim\Exception\MissingResponseFromRouteException
+     */
+    public function testEnforceReturn()
+    {
+        $callable = function ($req, $res, $args) {
+        };
+
+        $route = new Route(['GET'], '/', $callable);
+        $route->setEnforceReturn(true);
+        $route->setOutputBuffering(false);
```
Comment (akrabat) [inline tests/RouteTest.php:469]: Can we have a second test where this is the default please?

**Review by akrabat — CHANGES_REQUESTED** [review]
Looking good. Some minor bits and bobs noted.

File: tests/RouteTest.php [hunk 8]
```diff
@@ -479,8 +479,30 @@ public function testEnforceReturn()
...
+     * @expectedException \Slim\Exception\MissingResponseFromRouteException
+     */
+    public function testEnforceReturnWithOutputBuffering()
+    {
+        $callable = function ($req, $res, $args) {
+        };
+
+        $route = new Route(['GET'], '/', $callable);
+        $route->setEnforceReturnOfResponse(true);
+        $route->setOutputBuffering(true);
```
Comment (akrabat) [inline tests/RouteTest.php]: `true` is not a valid setting for outputBuffering.

File: Slim/App.php [hunk 9]
```diff
@@ -234,6 +234,7 @@ public function map(array $methods, $pattern, $callable)
             $callable = $callable->bindTo($this->container);
         }
 
+
```
Comment (dopesong) [inline Slim/App.php]: Double empty line?

**Comment by akrabat** [comment]
I've thought so more about this and while I appreciate that I asked for this, I now think it'd be better to wait for 4 and handle it better there.

**Comment by geggleto** [comment]
@akrabat do you want to leave the PR open or should be close it?

**Pull request closed without merging**